### PR TITLE
add statistics

### PR DESCRIPTION
**Pull request by ldorau**
This change is [Reviewable](https://reviewable.io/reviews/pmem/vmemcache/49)

### CODE REVIEW

**Comment by ldorau** [comment]
This PR contains PRs: #52 and #53

**Comment by ldorau** [comment]
This PR contains PR #52

**Comment by ldorau** [comment]
@pbalcer @kilobyte please review

File: src/libvmemcache.h [hunk 1]
```diff
@@ -90,6 +87,18 @@ enum vmemcache_replacement_policy {
 	VMEMCACHE_REPLACEMENT_NUM
 };
 
+enum vmemcache_statistics {
```
Comment (pbalcer) [inline src/libvmemcache.h]: drop the last `s` ;) `vmemcache_statistic`

File: src/vmemcache.c [hunk 2]
```diff
@@ -200,12 +200,17 @@ vmemcache_put(VMEMcache *cache, const char *key, size_t ksize,
...
+	size_t size_entry = sizeof(struct cache_entry) + ksize;
+
+	entry = Zalloc(size_entry);
 	if (entry == NULL) {
 		ERR("!Zalloc");
 		return -1;
 	}
 
+	entry->size = size_entry;
+	__sync_fetch_and_add(&cache->size_DRAM, entry->size);
```
Comment (pbalcer) [inline src/vmemcache.c]: util_fetch_and_add64 (same everywhere)

File: src/vmemcache.c [hunk 3]
```diff
@@ -336,6 +343,9 @@ vmemcache_entry_release(VMEMcache *cache, struct cache_entry *entry)
 	}
 
 	VEC_DELETE(&entry->value.fragments);
+
+	__sync_fetch_and_sub(&cache->size_DRAM, entry->size);
```
Comment (pbalcer) [inline src/vmemcache.c]: `malloc_usable_size(entry);`

File: src/libvmemcache.h (hunk 1 above)
Comment (ldorau) [reply src/libvmemcache.h]: Done

File: src/vmemcache.c (hunk 2 above)
Comment (ldorau) [reply src/vmemcache.c]: Done

File: src/vmemcache.c (hunk 3 above)
Comment (ldorau) [reply src/vmemcache.c]: Done

**Pull request merged**